components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1620,6 +1610,18 @@ mojom::AccountInfoPtr KeyringService::ImportAccountForKeyring(
   return account_info;
 }
 
+mojom::AccountInfoPtr KeyringService::GetHDAccountInfoForKeyring(
+    mojom::KeyringId keyring_id,
+    uint32_t index) const {
+  for (const auto& derived_account_info :
+       GetDerivedAccountsForKeyring(profile_prefs_, keyring_id)) {
+    if (derived_account_info.account_index == index) {
+      return MakeAccountInfoForDerivedAccount(derived_account_info);
+    }
+  }
+  return nullptr;
+}
```

Comment:
Let's not expose this.
Just do `keyring_service_->GetAllAccountInfos()` and find suitable account